### ISSUE TITLE
docs: Break out documentation to multiple files

### ISSUE THREAD

**Issue by onerandomusername** [DisnakeDev/disnake#392]
### Summary

Break out documentation to multiple files

### What is the feature request for?

The documentation

### The Problem

The docs can be slow, and they're a behemoth. There isn't much of a reason to have everything on the same page, except possibly for ease of search. 

### The Ideal Solution

We can probably make breakout pages but continue to build /api.html for everyone to use. This should be natively doable with sphinx and `include` directives.

This should not require us to duplicate any documentation, and should be easier to keep track of.

For example:

/api
/channel
/message
/utils

### The Current Solution

Wait for docs to load, often slowly.

**Comment by vincentwang0905** [comment]
I get stuck every time I open docs
Wait at least a minute or so

**Comment by ItsAleph** [comment]
I've got the first working prototype on local drive. All of the docs are moved to /api folder and split up by category (there isn't any rules, just my logical assumptions). Load times are much, much better, however, building docs from scratch takes longer for some reason (probably due to poor phone (yes im code from phone) processor, mid- and high-end users likely won't notice the difference). Also some "markers" (idk how they are called, im talking about those things in `conf.py`'s `rst_prolog` variable) were added to indicate whether the class/func is a discord model/data class/event etc. since there's no longer just one section for all models/data classes/events etc. yet we need the users know what is what. Didn't split ext.commands yet, and also consider ext.tasks too small to split up. I'll try to get my hands on making a draft pr soon, but it won't likely be ready in the next week or so because 9th school year and all..
